dashboard/utils_test.py

```diff
@@ -26,21 +25,21 @@ class MMTrackTest(TestCase):
     Tests for the MMTrack class
     """
 
+    # create Enrollments, Certificates, CurrentGrades
+    enrollments_json = load_json_from_file('dashboard/fixtures/user_enrollments.json')
+    certificates_json = load_json_from_file('dashboard/fixtures/certificates.json')
+    current_grades_json = load_json_from_file('dashboard/fixtures/current_grades.json')
+
     @classmethod
     def setUpTestData(cls):
         super().setUpTestData()
         # create an user
         cls.user = UserFactory.create()
 
-        # create Enrollments, Certificates, CurrentGrades
-        enrollments_json = load_json_from_file('dashboard/fixtures/user_enrollments.json')
-        cls.enrollments = Enrollments(enrollments_json)
-
-        certificates_json = load_json_from_file('dashboard/fixtures/certificates.json')
-        cls.certificates = Certificates([Certificate(cert_json) for cert_json in certificates_json])
-
-        current_grades_json = load_json_from_file('dashboard/fixtures/current_grades.json')
-        cls.current_grades = CurrentGrades([CurrentGrade(grade_json) for grade_json in current_grades_json])
+        with patch('dashboard.models.CachedEnrollment.data_qset', return_value=cls.enrollments_json), \
+                patch('dashboard.models.CachedCertificate.data_qset', return_value=cls.certificates_json), \
+                patch('dashboard.models.CachedCurrentGrade.data_qset', return_value=cls.current_grades_json):
```

Comment:
this is a somewhat tricky one. to start, here's a little snippet that actually mocks a queryset and gets us what we want for enrollments:

```python
from unittest.mock import MagicMock, patch
from dashboard.models import CachedEnrollment
mocked_enrollment_qset = MagicMock(spec=django.db.models.query.QuerySet, **{'__iter__.return_value': enrollments_json})
with patch('dashboard.models.CachedEnrollment.data_qset', return_value=mocked_enrollment_qset):
    qset = CachedEnrollment.data_qset(user=fake_user)
    print([x for x in qset])
```

it works but obviously a bit wonky. the purpose of this patch was to keep using the same fixtures that we have and avoid repeating the logic for creating edX objects for our raw data. i could also create actual cached records from the raw data, which would require that we pull the `edx_course_key` value for each object type, which would mean more repeated logic. i'm in between on this. how do you feel about the patch example?